File: ruby/rna-transcription/rna_transcription.rb
class Complement
  def self.of_dna(strand)
    rna = ''
    unless strand.match(/[^GTCA]/)
      strand.each_char do |nucleotide|
        case nucleotide
        when 'A'
          rna += 'U'
        when 'T'
          rna += 'A'
        when 'G'
          rna += 'C'
        when 'C'
          rna += 'G'
        end
      end
    end
    rna
  end
end

module BookKeeping
  VERSION = 4
end

File: php/rna-transcription/rna-transcription.php
<?php
function toRna($dna){
  $rna = '';
  for($i = 0; $i < strlen($dna); $i++){
    $nucleotide = $dna[$i];
    switch($nucleotide){
      case 'G':
        $rna .= 'C';
        break;
      case 'C':
        $rna .= 'G';
        break;
      case 'T':
        $rna .= 'A';
        break;
      case 'A':
        $rna .= 'U';
        break;
    }
  }
  return $rna;
}
?>
